app/lua/luac_cross/Makefile

```diff
@@ -75,7 +75,11 @@ endif
 
 all: $(DEPS) $(IMAGE)
 
-$(IMAGE) : $(OBJS)
+validate_lua_var:
+	@[ "$(LUA)" != 53 ] || { echo "E: Your variable LUA='$(LUA)' looks like you probably want app/lua53/host/Makefile instead."; exit 1; }
+	@[ -z "$(LUA)" -o "$(LUA)" = 51 ] || { echo "E: Unsupported setting LUA='$(LUA)'. For LUA 5.1, unset it or set it to empty or '51'."; exit 1; }
```

Comment:
This check is a bit overdone since AFAICS we don't check like that anywhere else either.

Reply:
AFAIR I added those error messages to give helpful messages for mistakes that I had done while trying, to save other people the effort to decipher the later failure messages. I strive to always improve user experience. If this makes other parts of the code inconsistent, I'd prefer we upgrade those other parts.

Reply:
Agreed. Then we should also add similar code in the lua53/host/Makefile and make sure that at least app\Makefile also fails if LUA contains something wrong.

Reply:
I'll try.

Reply:
Found two more places, and how to do the checks with Makefile commands.